OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -19,27 +19,26 @@
 namespace OpenRA.Mods.Common.Traits
 {
 	[Desc("This unit has access to build queues.")]
-	public class ProductionInfo : ITraitInfo
+	public class ProductionInfo : ConditionalTraitInfo
```

Comment:
We have the same enabled vs active distinction here. Consider #12665, which could be trivially fixed if we enabled the production queue on actor creation but kept it paused/inactive until the build animation has completed.